site/cds_rdm/ldap/utils.py

```diff
@@ -0,0 +1,119 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2023 CERN.
+#
+# CDS-RDM is free software; you can redistribute it and/or modify it under
+# the terms of the MIT License; see LICENSE file for more details.
+
+"""CDS-RDM ldap serializers."""
+
+from functools import partial
+
+from cds_rdm.ldap.errors import InvalidLdapUser
+from invenio_accounts.models import User
+from invenio_oauthclient.models import UserIdentity
+from invenio_userprofiles import UserProfile
+
+
+def serialize_ldap_user(ldap_user_data, log_func=None):
+    """Create ldap user."""
+
+    def serialize(ldap_user_data):
```

Comment:
several comments on the serialization:
- The two validation functions seem a bit redundant, since they check for the same. While the second one checks for an empty string, I think checking for `""` should also be part of the required because an empty email is a wrong response form the IdP/LDAP. Note that even the error message are identical.
- Then in lines 26-33 all keys on the dictionary are accessed with `[]` which means they are required (or it will raise a `KeyError`). Should they be accessed with `.get()`?